.sandbox/provision/lifecycle-asg/on-resume.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# : ${ASG_NAME:=test-windows-provision}
+# : ${AVAILABILITY_ZONE:=us-east-2a}
+# : ${EC2_SSH_KEY_FILE:=/run/sandbox/fs/secrets/shared/sandbox-shared.pem}
+
+set -ex
+
+source ./common.sh
+
+function on_resume() {
+    validate_asg "$ASG_NAME"
+    validate_az "$AVAILABILITY_ZONE"
+    validate_ssh_key "$EC2_SSH_KEY_FILE"
+
+    VOLUME_ID="$(get_volume_id)"
+
+    # TODO: need to lock and prevent race condition
+    INSTANCE_ID="$(claim_instance_from_asg)"
+    INSTANCE="$(aws ec2 describe-instances --instance-id "$INSTANCE_ID" --query 'Reservations[0].Instances[0]')"
+
+    PASSWORD="$(retrieve_password "$INSTANCE_ID" "$EC2_SSH_KEY_FILE")"
+    PUBLIC_DNS="$(echo "$INSTANCE" | jq -cMr .NetworkInterfaces[0].Association.PublicDnsName)"
+    PUBLIC_IP="$(echo "$INSTANCE" | jq -cMr .NetworkInterfaces[0].Association.PublicIp)"
+
+    attach_volume_if_needed "$VOLUME_ID" "$INSTANCE_ID"
+}
+
+on_resume >&2
+cat <<EOF
+{
+    "sandbox_id": "$SANDBOX_ID",
+    "sandbox_name": "$SANDBOX_NAME",
+    "password": "$PASSWORD",
+    "public_dns": "$PUBLIC_DNS",
+    "public_ip": "$PUBLIC_IP",
+    "volume_id": "$VOLUME_ID",
+    "instance": "$INSTANCE_ID"
+}
+EOF
```

Comment:
This has great similarity with `on-create.sh`, can we merged these by introducing an argument or checking the state or describe volume by tag if the volume has already been created.

Ideally, we can merge all these into one script `lifecycle.sh` and invoke using:

- `./lifecycle.sh create`
- `./lifecycle.sh suspend`
- `./lifecycle.sh resume`
- `./lifecycle.sh delete`